bridge/contracts/utils/auth/ImmutableALCABurner.sol

```diff
@@ -0,0 +1,33 @@
+// This file is auto-generated by hardhat generate-immutable-auth-contract task. DO NOT EDIT.
+// SPDX-License-Identifier: MIT-open-group
+pragma solidity ^0.8.16;
+
+import "contracts/utils/DeterministicAddress.sol";
+import "contracts/utils/auth/ImmutableFactory.sol";
+
+abstract contract ImmutableALCABurner is ImmutableFactory {
+    address private immutable _alcaBurner;
+    error OnlyALCABurner(address sender, address expected);
+
+    modifier onlyALCABurner() {
+        if (msg.sender != _alcaBurner) {
+            revert OnlyALCABurner(msg.sender, _alcaBurner);
+        }
+        _;
+    }
+
+    constructor() {
+        _alcaBurner = getMetamorphicContractAddress(
+            0x414c43414275726e657200000000000000000000000000000000000000000000,
```

Comment:
Comment magic numbers or use constant to add context.

Reply:
This will be changed in the following PR!